plugins/org.jboss.tools.quarkus.lsp4e/src/org/jboss/tools/quarkus/lsp4e/internal/ls/qute/QuteLanguageClient.java

```diff
@@ -104,10 +96,12 @@ public QuteLanguageClient() {
 	/**
 	 * 
 	 */
-	private void dispose() {
+	@Override
+	public void dispose() {
 		InstanceScope.INSTANCE.getNode(QuarkusLSPPlugin.PREFERENCES_QUALIFIER).removePreferenceChangeListener(this);
 		((IEclipsePreferences) Platform.getPreferencesService().getRootNode().node(ProjectScope.SCOPE)).removeNodeChangeListener(this);
 		preferencesNodes.forEach(node -> node.removePreferenceChangeListener(this));
+		super.dispose();
```

Comment:
Remove the listener inside the overrided dipose method:

`QutePlugin.getDefault().removeJavaDataModelChangedListener(listener);`